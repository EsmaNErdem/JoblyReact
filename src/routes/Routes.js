// Routes.js
import React from "react";
import { Route, Switch } from "react-router-dom";
import Home from "../homepage/Home";
import CompanyList from "../companies/CompanyList"
import CompanyDetail from "../companies/CompanyDetail";
import JobList from "../jobs/JobList";
import LoginForm from "../auth/LoginForm";
import SignUpFormForm from "../auth/SignUpForm";
import ProfileForm from "../profile/ProfileForm";

const Routes = () => {
  return (
    <Switch>
      <Route exact path="/">
        <Home />
      </Route>
      <Route exact path="/companies">
        <CompanyList />
      </Route>
      <Route path="/companies/:handle">
        <CompanyDetail />
      </Route>
      <Route exact path="/jobs">
        <JobList />
      </Route>
      <Route exact path="/profile">
        <ProfileForm />
      </Route>
      <Route exact path="/login">
        <LoginForm />
      </Route>
      <Route exact path="/signup">
        <SignUpFormForm />
      </Route>
    </Switch>
  );
};

export default Routes;
