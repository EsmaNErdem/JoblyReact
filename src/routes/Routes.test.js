import { render, cleanup } from "@testing-library/react";
import Routes from "./Routes";
import { MemoryRouter } from "react-router";
import { UserProvider, NonUserProvider } from "../testUtils";

jest.mock('../api');
afterEach(cleanup);

test("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <Routes />
        </UserProvider>
      </MemoryRouter>,
  );
});

test("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <Routes />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

test("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <NonUserProvider>
          <Routes />
        </NonUserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
