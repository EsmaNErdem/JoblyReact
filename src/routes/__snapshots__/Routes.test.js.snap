// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      Jobly
    </h1>
    <p>
      Find yourself a job
    </p>
    <h2>
      Welcome, testuser!
    </h2>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="Home"
  >
    <h1>
      Jobly
    </h1>
    <p>
      Find yourself a job
    </p>
    <p>
      <a
        href="/login"
      >
        Login
      </a>
      <a
        href="/signup"
      >
        Sign Up
      </a>
    </p>
  </div>
</DocumentFragment>
`;
